### PR TITLE
maint: refactor to extract fiddly logic into modules

### PR DESCRIPTION
**Pull request by robbkidd**
I propose that the web application is easier to read when the fiddly details of how things are retrieved live in named modules. This makes the apps look more like Real Code™ without making them wildly complicated.

### CODE REVIEW

File: services/meminator-ruby/src/app.rb [hunk 1]
```diff
@@ -11,66 +9,12 @@ class MeminatorApp < Sinatra::Application
...
-      conversion_span.set_attribute("app.subprocess.command", memination_command)
-
-      cmd_out, cmd_err, status = Open3.capture3(memination_command)
-      conversion_span.add_attributes({
-        "app.subprocess.returncode" => status.exitstatus,
-        "app.subprocess.stderr" => cmd_err,
-        "app.subprocess.stdout" => cmd_out,
-      })
+    phrase, image_url, errors = MemeGenerator.parse_request(request)
+    return [400, { errors: errors}.to_json] if !errors.empty?
```
Comment (dmarchante) [inline services/meminator-ruby/src/app.rb:14]: Are we not losing some of the custom instrumentation in this pattern?

Comment (robbkidd) [reply services/meminator-ruby/src/app.rb:14]: I did remove the custom attributes and the custom convert span in this PR.

I could add them back in, but there's a tickle in my memory now that a student would be guided towards adding these custom instruments around the long download from S3 and entirely-opaque-to-auto-instrumentation shell out command.

I can't find the reference now, but in my head it is:

1. learn how to add auto-instrumentation (to backend-for-frontend)
2. learn how to add attributes to existing auto-instrumentation spans

Now that the student can see a single trace …
* Notice how meminator takes a long time? 
* What's up with that?
* Oh, there's a long call to an external service.
* Double-oh, there's this shell-out that I don't get a span for.

3. learn how to add custom spans around your own interesting business logic

Am I off the Rails† on this?

--- 

† See what I did there?

Comment (robbkidd) [reply services/meminator-ruby/src/app.rb:14]: @dmarchante and I chatted.

I'll add the `convert` span around the shell-out command for that to exist as the student's starting state. Scripts exist for other languages and those don't have the student create a custom span in the meminator service.

**Review by tdarwin — APPROVED** [review]
This looks better to me, though the only thought I had is that the `backend-for-frontend` service will look a bit more hectic with an `app.rb`, `client.rb`, and `observability.rb` post instrumentation

**Comment by robbkidd** [comment]
Merging to get these changes in. If we'd like to make further changes, we'll open more PRs.

**Pull request merged**